backend/logic/order.js

```diff
@@ -0,0 +1,160 @@
+const randomstring = require('randomstring')
+const util = require('ethereumjs-util')
+
+const { OrderStatuses } = require('../enums')
+const { Sentry } = require('../sentry')
+const { Order } = require('../models')
+const sendNewOrderEmail = require('../utils/emails/newOrder')
+const discordWebhook = require('../utils/discordWebhook')
+const { autoFulfillOrder } = require('../utils/printful')
+const { decryptShopOfferData } = require('../utils/offer')
+const { validateDiscountOnOrder } = require('../utils/discounts')
+const { getLogger } = require('../utils/logger')
+
+const log = getLogger('logic.order')
+
+/**
+ * Create a new unique order ID that is human readable and can be used by
+ * the buyer and the seller to reference to a specific order.
+ * Format: <networkId>-<shopId>-<randomId>.
+ * Example: 1-123-XCQ69BTJ
+ *
+ * @param {models.Shop} shop
+ * @returns {string}
+ */
+function createNewOrderId(shop) {
+  const randomId = randomstring.generate({
+    readable: true,
+    charset: 'alphanumeric',
+    capitalization: 'uppercase',
+    length: 8
+  })
+  return `${shop.networkId}-${shop.id}-${randomId}`
```

Comment:
If we decide to use listingId, I'd vote for always using the server listingId. So that in the future we don't require a listingId when a merchant sets up a new shop (that's a high friction point).

A downside is that it will add an extra step (e.g. sending a transaction to create a listing + wait for it to get mined) when a new deployer system gets setup (ex: via GCP marketplace). But that's probably ok given one could expect that a deployer setup is a fairly technical process.

Reply:
We decided to use <netId>-<contractVersion>-<listingId>-<shopId>-<randomId> for the fully qualified form.
For the short form used as a reference number for external communication, we'll just use <randomId>.